File: 6.DP/BOJ/(18353번)병사 배치하기/[jeunghun] solution.py
n = int(input())
data = list(map(int, input().split()))
dp = [1] * n
max_value = 1

for i in range(n - 1, 0, -1):
    for j in range(i - 1, -1, -1):
        if data[i] < data[j]:
            dp[j] = max(dp[i] + 1, dp[j])

            if dp[j] > max_value:
                max_value = dp[j]

print(n - max_value)

File: 6.DP/BOJ/(18353번)병사 배치하기/[jeunghun] solution.swift
let n = Int(readLine()!)!
let data = readLine()!.split(separator: " ").map { Int($0)! }
var dp = Array(repeating: 1, count: n)

for i in 0..<n - 1 {
    for j in i + 1..<n {
        if data[i] > data[j] {
            dp[j] = max(dp[i] + 1, dp[j])
        }
    }
}

print(n - dp.max()!)